Snowflake.Data.Tests/IntegrationTests/SFPutGetTest.cs

```diff
@@ -496,23 +513,48 @@ private void GetFile(DbConnection conn)
                     reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.ResultStatus));
 
                 // Check file contents
-                using (var streamReader = new StreamReader(t_outputFilePath))
+                foreach (var line in ReadOutputFileLinesUnzippingIfReallyCompressed())
                 {
-                    while (!streamReader.EndOfStream)
+                    if (string.IsNullOrEmpty(line)) continue;
+                    var values = line.Split(',');
+
+                    for (var i = 0; i < s_colData.Length; i++)
                     {
-                        var line = streamReader.ReadLine();
-                        if (line == null) continue;
-                        var values = line.Split(',');
-
-                        for (var i = 0; i < s_colData.Length; i++)
-                        {
-                            Assert.AreEqual(s_colData[i], values[i]);
-                        }
+                        Assert.AreEqual(s_colData[i], values[i]);
                     }
                 }
             }
         }
 
+        private static string[] ReadOutputFileLinesUnzippingIfReallyCompressed()
+        {
+            using (var outputStream = File.OpenRead(t_outputFilePath))
+            {
+                using (var processedOutputStream = IsCompressedByTheDriver() ? Unzip(outputStream) : outputStream)
```

Comment:
Am I right that we do not compress files according to t_sourceCompressionType but only change the extension?
If so it means we do not test PUT-ing really compressed files. A little pity.

Reply:
exactly, the files have only extensions like .gzip indicating that they are compressed, but they are not.